lib/src/utils/formatter_inputs.dart

```diff
@@ -0,0 +1,75 @@
+import 'dart:io';
+
+import 'package:glob/glob.dart';
+import 'package:path/path.dart' as p;
+
+/// Builds and returns the object that contains:
+/// - The file paths
+/// - The paths that were excluded by an exclude glob
+/// - The paths that were skipped because they are links
+/// - The hidden directories(start with a '.') that were skipped
+///
+/// The file paths will include all .dart files in [path] (and its subdirectories),
+/// except any paths that match the expanded [exclude] globs.
+///
+/// By default these globs are assumed to be relative to the current working
+/// directory, but that can be overridden via [root] for testing purposes.
+FormatterInputs getFormatterInputs({List<Glob> exclude, String root}) {
+  FormatterInputs inputs = FormatterInputs(Set(), Set(), Set(), Set());
```

Comment:
It's an anti-pattern to create an object with collection instances and then later mutate those collections. What I was going for with making the fields on the `FormatterInputs` class final was to ensure that the returned object from this function is immutable. I think a better pattern would be to create the `Set`s at the top of this function, and then construct the `FormatterInputs` object at the end when we return, e.g.:

```dart
final included = <String>{};
final excluded = <String>{};
final links = <String>{};
final hidden = <String>{};

...

return FormatterInputs(
  included,
  excludedFiles: excludedFiles,
  hiddenDirectories: hiddenDirectories,
  skippedLinks: skippedLinks);
```

(Also, the dart style guide now recommends using the collection literals when possible: https://dart.dev/guides/language/effective-dart/usage#do-use-collection-literals-when-possible)